### ISSUE TITLE
AGR Animations flipping, and created more instances than needed.

### ISSUE THREAD

**Issue by ghost** [advancedfx/advancedfx#246]
I launched CSGO with HLAE using these settings:

 -console -insecure -sv_lan 1 -console +mat_wueue_mode 2 -window

When I record in CSGO, I used the following settings, pulled from the devomvm config: 

host_timescale 0; host_framerate 30; mirv_streams record agr enabled 1; mirv_streams record agr recordCamera 1; mirv_streams record agr recordPlayers 1;  mirv_streams record agr recordWeapons 0; mirv_streams record agr recordInvisible 1; mirv_streams record agr recordProjectiles 0; mirv_streams record agr recordViewmodel 0

After the record and import to blender, I end up with many objects and hierarchies that do not belong there. As in Devostated's demonstration, he only has about a quarter of the objects that I do, and we recorded the same things. In addition to this, the ragdoll animations are flipped, so they do not line up. Lastly, as in one of the pictures I have attached, it shows a 3rd model being added to the location of the others, seeming to have no correlation whatsoever with what is happening with the animations.

The .agr .dem and .blend files are included in a zip file on my google drive:
https://drive.google.com/open?id=1_O4j8ITERmIrgXk-dX6dwdQCRmWQkSmL

Thank you for your help!

Contact me on discord if you need additional information:
𝓛𝓾𝓶𝓲𝓷#3849

![Random3rd model](https://user-images.githubusercontent.com/37248926/68089305-516abd80-fe1c-11e9-973d-c87c0546f166.PNG)
![FlippedRagdoll](https://user-images.githubusercontent.com/37248926/68089324-8a0a9700-fe1c-11e9-8895-63924747b1f4.PNG)

**Comment by dtugend** [comment]
> "I end up with many objects and hierarchies that do not belong there. As in Devostated's demonstration, he only has about a quarter of the objects that I do, and we recorded the same things

He probably had recordInvisible 0, in that cases the import script can merge more objects together and also less objects are recorded. Extending that is more of a feature request though, it probably won't happen before HLAE v3.

> In addition to this, the ragdoll animations are flipped, so they do not line up.

This is a CS:GO bug that happens in old demos that were before a specific CS:GO update, it's not caused by the AGR / HLAE recording, if you look closely in-game it will happen there too with the same demo.
I tried various things to fix it in-game for old demos, but after quite some time spent on that, I decided to not try any further, especially since newer demos are fine.

> it shows a 3rd model being added to the location of the others, seeming to have no correlation whatsoever with what is happening with the animations.

This happens in GOTV demos, it's you the spectator that recorded the AGR, it happens because of recordInvisible 0. just delete it.
